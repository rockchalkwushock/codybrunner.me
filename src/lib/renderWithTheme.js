import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from './theme'

const renderWithTheme = Component =>
  mount(<ThemeProvider theme={theme}>{Component}</ThemeProvider>)

export default renderWithTheme
