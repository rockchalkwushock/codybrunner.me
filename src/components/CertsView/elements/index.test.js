import React from 'react'
import { shallow } from 'enzyme'

import { CertsFlexContainer, CertsGrid } from './'
import renderWithTheme from '../../../lib/renderWithTheme'

describe('Component: <CertsFlexContainer />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<CertsFlexContainer />)
  })

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should have styles', () => {
    expect(wrapper).toHaveStyleRule('align-items', 'unset')
    expect(wrapper).toHaveStyleRule('display', 'flex')
    expect(wrapper).toHaveStyleRule('flex-direction', 'column')
    expect(wrapper).toHaveStyleRule('justify-content', 'center')
    expect(wrapper).toHaveStyleRule('text-align', 'center')
  })
})

describe('Component: <CertsGrid />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = renderWithTheme(<CertsGrid />)
  })

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should have styles', () => {
    expect(wrapper).toHaveStyleRule('background-color', 'inherit')
    expect(wrapper).toHaveStyleRule('border-radius', '10px')
    expect(wrapper).toHaveStyleRule(
      'box-shadow',
      '1px 1px 5px 1px rgba(0,0,0,0.7)'
    )
    expect(wrapper).toHaveStyleRule('display', 'grid')
    expect(wrapper).toHaveStyleRule('grid-gap', '10px')
    expect(wrapper).toHaveStyleRule(
      'grid-template-columns',
      'repeat(auto-fit,minmax(275px,1fr))'
    )
    expect(wrapper).toHaveStyleRule('justify-items', 'center')
    expect(wrapper).toHaveStyleRule('margin', '1em 0')
    expect(wrapper).toHaveStyleRule('padding', '2em 1em')
  })
})
