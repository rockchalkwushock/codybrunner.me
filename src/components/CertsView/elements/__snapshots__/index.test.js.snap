// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <CertsFlexContainer /> should render correctly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: unset;
  -webkit-box-align: unset;
  -ms-flex-align: unset;
  align-items: unset;
  text-align: center;
}

<div
  className="c0"
/>
`;

exports[`Component: <CertsGrid /> should render correctly 1`] = `
.c0 {
  background-color: inherit;
  border-radius: 10px;
  box-shadow: 1px 1px 5px 1px rgba(0,0,0,0.7);
  display: grid;
  grid-gap: 10px;
  grid-template-columns: 1fr;
  justify-items: center;
  margin: 1em 0;
  padding: 2em 1em;
  grid-template-columns: repeat(auto-fit,minmax(275px,1fr));
}

<ThemeProvider
  theme={
    Object {
      "post": Object {
        "bg": "#f9f8eb",
        "ftClr": "#000",
        "ftFm": "\\"Raleway\\", sans-serif",
        "ftSz": "1.1rem",
        "ftWt": "400",
        "linkClr": "#143a52",
        "linkHvClr": "#7a9eb1",
        "linkWt": "600",
        "lnHt": "1.618",
        "ltSp": "1px",
      },
      "screen": Object {
        "large": "736px",
      },
      "site": Object {
        "bg": "#7a9eb1",
        "borderRadius": "10px",
        "fg": "#ffe1b6",
        "ftClr": "#143a52",
        "ftFm": "\\"Raleway\\", sans-serif",
        "ftSz": "1rem",
        "ftWt": "400",
        "linkClr": "#143a52",
        "linkHvClr": "#ffe1b6",
        "lnHt": "1.4",
        "ltSp": "0px",
        "shadow": "1px 1px 5px 1px rgba(0, 0, 0, 0.7)",
        "transition": "all 0.3s ease-in-out",
      },
    }
  }
>
  <CertsGrid>
    <div
      className="c0"
    />
  </CertsGrid>
</ThemeProvider>
`;
